### PR TITLE
CLOUDSTACK-5806: add presetup to storage types that support over prov…

### PR DESCRIPTION
**Pull request by abhinandanprateek**
…isioning

Ideally this should be configurable via global settings

### CODE REVIEW

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by borisstoyanov** [comment]
@abhinandanprateek I think we hit an issue with the test: 

```
  File "/marvin/tests/smoke/test_over_provisioning.py", line 83, in test_UpdateStorageOverProvisioningFactor

CloudstackAPIException: Execute cmd: updateconfiguration failed, due to: errorCode: 431, errorText:Unable to update  storage pool with id 2. Overprovision not supported for NetworkFilesystem
-------------------- >> begin captured stdout << ---------------------
=== TestName: test_UpdateStorageOverProvisioningFactor | Status : EXCEPTION ===
```
Reading the code changes looks like over provisioning is not supported for NFS, if that's expected we should edit the test I think.

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 1]
```diff
@@ -508,7 +507,7 @@ public String updateConfiguration(final long userId, final String name, final St
                     throw new InvalidParameterValueException("unable to find storage pool by id " + resourceId);
                 }
                 if(name.equals(CapacityManager.StorageOverprovisioningFactor.key())) {
-                    if(pool.getPoolType() != StoragePoolType.NetworkFilesystem && pool.getPoolType() != StoragePoolType.VMFS) {
+                    if(pool.getPoolType().supportsOverProvisioning() ) {
```
Comment (serg38) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: I believe it should be negated

Comment (abhinandanprateek) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: yep ! thanks @serg38

**Comment by abhinandanprateek** [comment]
@borisstoyanov a condition got inverted by the change (serg38 pointed that out above), it is fixed now. Can you rerun the tests.

**Comment by borisstoyanov** [comment]
Thanks @abhinandanprateek, 
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by borisstoyanov** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
LGTM.

**Pull request merged**